cfme/v2v/migrations.py

```diff
@@ -402,28 +474,77 @@ def create(self, form_data):
         view.form.fill(form_data)
         return infra_map
 
-# TODO: Next Entity and Collection classes are to be filled by Yadnyawalk(ytale),
-# which he will submit PR for once my PR merged.
-
 
 @attr.s
 class MigrationPlan(BaseEntity):
     """Class representing v2v Migration Plan"""
-    # TODO: Ytale is updating rest of the code in this entity in separate PR.
-    category = 'migrationplan'
-    string_name = 'Migration Plan'
-    name = 'plan1'
+    name = attr.ib()
 
 
 @attr.s
 class MigrationPlanCollection(BaseCollection):
-    """Collection object for Migration Plan object"""
-    # TODO: Ytale is updating rest of the code in this collection in separate PR.
+    """Collection object for migration plan object"""
     ENTITY = MigrationPlan
 
+    def create(self, name, infra_map, vm_names, description=None, csv_import=False,
+               start_migration=False):
+        """Create new migration plan in UI
+        Args:
+            name: (string) plan name
+            description: (string) plan description
+            infra_map: (object) infra map object name
+            vm_names: (list) vm names
+            csv_import: (bool) flag for importing vms
+            start_migration: (bool) flag for start migration
+        """
+        view = navigate_to(self, 'Add')
+        view.general.fill({
+            'infra_map': infra_map,
+            'name': name,
+            'description': description
+        })
+
+        if csv_import:
+            view.general.select_vm.select("Import a CSV file with a list of VMs to be migrated")
+            view.next_btn.click()
+            temp_file = tempfile.NamedTemporaryFile(suffix='.csv')
+            with open(temp_file.name, 'w') as file:
+                headers = ['Name', 'Provider']
+                writer = csv.DictWriter(file, fieldnames=headers)
+                writer.writeheader()
+                for vm in vm_names:
+                    writer.writerow({'Name': vm.name, 'Provider': vm.provider.name})
+            view.vms.hidden_field.fill(temp_file.name)
+        else:
+            view.next_btn.click()
+
+        wait_for(lambda: view.vms.table.is_displayed, timeout=60, message='Wait for VMs view',
+                 delay=2)
+        for vm in vm_names:
+            view.vms.filter_by_name(vm.name)
+            for row in view.vms.table.rows():
+                if row['VM Name'].read() in vm_names:
+                    row['Select'].fill(True)
+            view.vms.clear_filters.click()
+        view.next_btn.click()
+
+        if start_migration:
+            view.options.run_migration.select("Start migration immediately")
+        view.options.create_btn.click()
+        wait_for(lambda: view.results.msg.is_displayed, timeout=60, message='Wait for Results view')
+
+        base_flash = "Migration Plan: '{}'".format(name)
```

Comment:
Would rather see this message in the CRUD test than here in the create method. Just `assert_no_error` here in create, so changing flash text won't break tests that just use migration plans for setup.